Makefile

```diff
@@ -1,6 +1,6 @@
 CC ?= gcc
-CFLAGS = -Wall -O2
-LDFLAGS = -lm
+CFLAGS = -Wall -O2 -march=native -fopenmp
```

Comment:
Let's allow conditional builds. That is, we can rewrite as following:
```
CFLAGS = -Wall -O2
LDFLAGS = -lm

ifeq ("$(ENABLE_OPENMP)","1")
CFLAGS += -DENABLE_OPENMP -march=native -fopenmp
LDFLAGS += -fopenmp -lpthread
endif
```

Reply:
done